docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,193 @@
+[Workflows](/workers). On Task Queues using this feature, the Workflow starter does not need to know
+when new versions are introduced. New [Workflow Executions](/workflows#workflow-execution) run using
+the new code in the newly deployed Workers, and old Workflow Executions are processed only by
+Workers with an appropriate version. Old workers can be decommissioned after the archival of any
+open workflows using their version, or if you do not care about querying closed workflows, once
+there are no longer any open workflows at that version.
+
+For example, if you have a Workflow that never takes longer than a day to execute, a great strategy
+is to always assign a new build identifier to every new build of your Worker and add it to the
+version sets as the new overall default. Because you know that your Workflow never takes more than a
+day to complete, you know that you'll never have to leave older Workers running for more than a day
+after you deploy the new version (assuming availability). You don't need to worry about removing old
+versions from the sets. Leaving them there is harmless.
+
+You can use this technique with longer-lived Workflows as well. Just be aware that you might need to
+keep multiple versions of Workers running while open Workflows complete.
+
+You can also use the feature to prevent a buggy codepath from being taken on currently open
+Workflows. You can do this by adding a new version to an existing set and specifying it as
+_compatible_ with an existing version on which no future Workflow Tasks should execute. Because the
+new version processes existing [Event Histories](/workflows/#event-history), it is subject to the
+normal [deterministic constraints](/workflows/#deterministic-constraints), and you might need to use
+one of the [versioning APIs](/workflows/#workflow-versioning).
+
+The feature also permits you to make incompatible changes to Activity Definitions in conjunction
+with incompatible changes to Workflow Definitions that use those Activities. This works because any
+Activity scheduled by a Workfow on the same Task Queue will, by default, only be dispatched to
+Workers that have a compatible build ID with the Workflow that scheduled it. Thus, if you wish
+to change the type signature of an Activity Definition while creating a new incompatible build ID
+for a Worker, you can do so without concern that the Activity will fail to execute on some other
+Worker with an incompatible definition. The same principle applies to Child Workflows.
+
+Note that if you schedule an Activity or a Child Workflow on _different_ Task Queue from the one
+the Workflow is running on, they will be not be assigned any particular version.
+
+Continue-as-new from a Workflow on a versioned Task Queue will, by default, start the continued
+Workflow on the same compatible set as the original Workflow. Continuing-as-new onto a different
+task queue will not assign any particular version.
```

Comment:
Yes... but I wanted to avoid having to specify exactly how a user does that b/c we haven't added it to the command yet and it'll probably look slightly different in each SDK. I could say something generic like "Look at your SDK's reference docs for c-a-n" or something

Reply:
I think we're going to need this feature for child workflows as well.
This default behavior where everything stays in the same compatible version will make it hard to retire old workers in some scenarios.

Reply:
Left it generic for now